src/generator/models.ts

```diff
@@ -215,56 +211,32 @@ function determineMarshallersForObj(obj: ObjectSchema): Marshallers {
 }
 
 // walks the inheritance graph of obj to determine marshallers.
-// when parents is true, the parents are walked, else the children.
-function recursiveDetermineMarshallers(obj: ObjectSchema, parents: boolean): Marshallers {
```

Comment:
I changed this to a classic depth-first tree traversal algorithm which is what it should have been in the first place.